### PR TITLE
Domain Restore OpenAPI spec

### PR DESCRIPTION
**Pull request by OleMchls**
As part of the ongoing restore project, this PR adds the Open API specs for the restore API.

### CODE REVIEW

**Review by kntsoriano — APPROVED** [review]

File: content/v2/openapi.yml [hunk 1]
```diff
@@ -1178,6 +1179,56 @@ paths:
...
+          description: Domain restore created
+          content:
+            application/json:
+              schema:
+                type: object
+                properties:
+                  data:
+                    $ref: '#/components/schemas/DomainRestore'
+        '400':
+          $ref: '#/components/responses/400'
```
Comment (DXTimer) [inline content/v2/openapi.yml:1207]: Wouldn't 404 and 401 also be valid response codes for this endpoint?

File: content/v2/openapi.yml [hunk 2]
```diff
@@ -1178,6 +1179,56 @@ paths:
...
+          description: Domain restore details.
+          content:
+            application/json:
+              schema:
+                type: object
+                properties:
+                  data:
+                    $ref: '#/components/schemas/DomainRestore'
+        '404':
+          $ref: '#/components/responses/404'
```
Comment (DXTimer) [inline content/v2/openapi.yml:1231]: Similarly wouldn't 401 be a valid response code?

**Review by DXTimer — APPROVED** [review]
Left some comments otherwise it looks great. :shipit:

File: content/v2/openapi.yml (hunk 2 above)
Comment (OleMchls) [reply content/v2/openapi.yml:1237]: Added in 16530de7e8d1199fb4e2332bc33a4b3d336f0cb9

**Pull request merged**